README.md
# Simple-Coins-0.2.0
The two coins serve no special purpose but to enhance the Vanilla trading experience in Minecraft, more specifically in multiplayer. It provides more options to use as currency. You can also craft back to an iron or gold ingot by putting 4 coins (of the same variant) into a 2x2 square.


Changelog.txt
1.0.0
	Bankers now have their own house. As such, they can now natively spawn in villages.
0.7.0
	Coins now generate in loot chests.
		0-5 iron and gold coins can generate in all types randomly generated tests (including the starter chest).
	Added a new Configuration Category: World Generation. Set the values for coins generating across the world.
0.6.0
	Added a Configuration menu.
		Enable/Disable Banker Villager.
		Enable/Disable Tier 1 and 2 Trades.
		Set the min/max for #Coins <-> 1 Emerald for Tier 1 and 2.
		Can change crafting amount for a coin and how many coins are produced.
		Set drop rates for coins on mobs.
		Set min/max amount of coins that mobs can drop.
		Enable/Disable chance of drops for some mobs.
0.5.2
	Banker now named "Banker" in en_US.lang.
	Figured out trades for 1 emerald to multiple items.
	At Tier 1:
		21-27 Iron Coins <-> 1 Emerald
		24-30 Gold Coins <-> 1 Emerald
	At Tier 2:
		21-27 Iron Coins -> 1 Emerald
		24-30 Gold Coins -> 1 Emerald
		38-50 Iron Coins <- 2 Emeralds
		24-30 Gold Coins <- 2 Emeralds
0.5.1
	Emeralds cannot be converted to coins properly. This trade option has been removed for the time being.
	That means this trade is what is left:
	At Tier 1:
		21-27 Iron Coins -> 1 Emerald
		24-30 Gold Coins -> 1 Emerald
	At Tier 2:
		21-27 Iron Coins -> 1 Emerald
		24-30 Gold Coins -> 1 Emerald
0.5.0
	Added new type of Villager: Banker.
	Banker wears a light blue robe.
	At Tier 1:
		21-27 Iron Coins <-> 1 Emerald
		24-30 Gold Coins <-> 1 Emerald
	At Tier 2:
		21-27 Iron Coins -> 1 Emerald
		24-30 Gold Coins -> 1 Emerald
		38-50 Iron Coins <- 2 Emeralds
		24-30 Gold Coins <- 2 Emeralds
0.4.2
	3 Nuggets are required for 1 coin.
	1 coin back crafts to 3 nuggets.
	Fixed a crafting bug that would not waste nuggets and not give you a coin.
0.4.0 - Coins added to mob drops.
	Zombies, ZombiePigmen, Husks, and ZombieVillagers have a 20% chance of
	dropping 4 iron coins, and a 10% chance of dropping 4 coins.
	Evokers, Vindicators and Villagers all drop 0-3 iron coins and 0-3 gold
	coins
	Updated to Forge 13.20.0.2212
0.3.1
	Updated to Forge 13.20.0.2206
0.3.0
	Mint Block is now functionality. Can craft 1 coin from 1 nugget (Iron or Gold).
	Recipes for coins in the crafting table has been removed. THE MINT IS NOW REQUIRED TO CRAFT COINS
	Mint Block crafting recipe changed. Check the crafting recipes above.
	1 Coin in the crafting table now becomes 1 nugget.
0.2.0 - Initial Release.
	Mint block added.
	Iron and Gold coin added.

src/java/com/TonyTiger/simplecoins/config/ConfigHandler.java
package com.TonyTiger.simplecoins.config;

import java.io.File;

import com.TonyTiger.simplecoins.ClientProxy;
import com.TonyTiger.simplecoins.event.EventHandlerCommon;
import com.TonyTiger.simplecoins.guicontainer.ContainerMintTileEntity;
import com.TonyTiger.simplecoins.profession.BankVillageHandler;
import com.TonyTiger.simplecoins.profession.BankerProfessionAndCareer;
import com.TonyTiger.simplecoins.world.genCoinsInWorld;

import net.minecraftforge.common.config.Configuration;

public class ConfigHandler {
	public static Configuration configMain = 
			new Configuration(new File("config/simplecoins.cfg"));
	public static void syncConfig(){
		init();
		if(configMain.hasChanged())
			configMain.save();
	}
	
	public static void init(){
		String category = "";
		
		category = "Villagers";
		configMain.addCustomCategoryComment(category, "Village Settings");
		ClientProxy.enableBanker = configMain.getBoolean("Enable Banker", category, true, "Villager that allows trading to and from emeralds with coins. Comes with it's own building.");
		BankVillageHandler.limit = configMain.getInt("Max amount of Banks in a Village", category, 1, 0, 10, "Adjust the amount of banks that can spawn in a village.");
		
		category = "Trading";
		configMain.addCustomCategoryComment(category, "Trading Settings");
		BankerProfessionAndCareer.enableTier1trades = configMain.getBoolean("Enable Tier 1 Trades", category, true, "Enable/Disable first tier of trades. I recommend you turn them on");
		BankerProfessionAndCareer.tier1IronMin = configMain.getInt("T1: Iron Min", category, 21, 1, 64, "Lowest amount of iron coins that can be trade for 1 emerald. Must be lower or the same to its max.| (1,64) | 1st Tier");
		BankerProfessionAndCareer.tier1IronMax = configMain.getInt("T1: Iron Max", category, 27, 1, 64, "Highest amount of iron coins that can be trade for 1 emerald. Must be more or the same to its min.| (1,64) | 1st Tier");
		BankerProfessionAndCareer.tier1GoldMin = configMain.getInt("T1: Gold Min", category, 24, 1, 64, "Lowest amount of gold coins that can be trade for 1 emerald. Must be lower or the same to its max.| (1,64) | 1st Tier");
		BankerProfessionAndCareer.tier1GoldMax = configMain.getInt("T1: Gold Max", category, 30, 1, 64, "Highest amount of gold coins that can be trade for 1 emerald. Must be more or the same to its min.| (1,64) | 1st Tier");
		BankerProfessionAndCareer.enableTier2trades = configMain.getBoolean("Enable Tier 2 trades", category, false, "Enable/Disable second tier of trades. Default is false. No recommendation, up to you if you want tier 2 trades.");
		BankerProfessionAndCareer.tier2IronMin = configMain.getInt("T2: Iron Min", category, 19, 1, 32, "Lowest amount of iron coins that can be trade for 1 emerald. Must be lower or the same to its max.| (1,32) | 2nd Tier");
		BankerProfessionAndCareer.tier2IronMax = configMain.getInt("T2: Iron Max", category, 25, 1, 32, "Highest amount of iron coins that can be trade for 1 emerald. Must be more or the same to its min.| (1,32) | 2nd Tier");
		BankerProfessionAndCareer.tier2GoldMin = configMain.getInt("T2: Gold Min", category, 22, 1, 32, "Lowest amount of gold coins that can be trade for 1 emerald. Must be lower or the same to its max.| (1,32) | 2nd Tier");
		BankerProfessionAndCareer.tier2GoldMax = configMain.getInt("T2: Gold Max", category, 28, 1, 32, "Highest amount of gold coins that can be trade for 1 emerald. Must be more or the same to its min.| (1,32) | 2nd Tier");
		
		category = "Crafting";
		configMain.addCustomCategoryComment(category, "Mint Crafting Settings");
		ContainerMintTileEntity.numCoinsCrafted = configMain.getInt("Number of Coins Crafted", category, 1, 1, 64, "How many coins are crafted in the Mint.");
		ContainerMintTileEntity.numNuggetsNeedForCraft = configMain.getInt("Nuggets need to craft Coins", category, 3, 1, 64, "How many nuggest are needed in the Mint.");
		
		
		category = "Mob Drops";
		configMain.addCustomCategoryComment(category, "Mob Drop Settings");
		EventHandlerCommon.ironcoinDroprate = (float)(configMain.getInt("Iron Coins Drop Rate", category, 5, 0, 100, "Chance of Iron Coins dropping in the mobs below.")/100.0);
		EventHandlerCommon.goldcoinDroprate = (float)(configMain.getInt("Gold Coins Drop Rate", category, 10, 0, 100, "Chance of Gold Coins dropping in the mobs below.")/100.0);
		EventHandlerCommon.goldcoinMinDropAmt = configMain.getInt("Minimum Amount of Gold Drop", category, 0, 0, 64, "If higher than maximum, the max will be set to this value.");
		EventHandlerCommon.goldcoinMaxDropAmt = configMain.getInt("Maximum Amount of Gold Drop", category, 4, 0, 64, "If lower than minimum, it will be equal to the min value set.");
		EventHandlerCommon.ironcoinMinDropAmt = configMain.getInt("Minimum Amount of Iron Drop", category, 0, 0, 64, "If higher than maximum, the max will be set to this value.");
		EventHandlerCommon.ironcoinMaxDropAmt = configMain.getInt("Maximum Amount of Iron Drop", category, 4, 0, 64, "If lower than minimum, it will be equal to the min value set.");
		EventHandlerCommon.enableEvokerdrops = configMain.getBoolean("Enable Evoker Drops", category, true, "");
		EventHandlerCommon.enableHuskdrops = configMain.getBoolean("Enable Husk Drops", category, true, "");
		EventHandlerCommon.enablePigZomdrops = configMain.getBoolean("Enable Zombie Pigman Drops", category, true, "");
		EventHandlerCommon.enableVillagerdrops = configMain.getBoolean("Enable Villager Drops", category, true, "");
		EventHandlerCommon.enableVindicatordrops = configMain.getBoolean("Enable Vindicator Drops", category, true, "");
		EventHandlerCommon.enableZombiedrops = configMain.getBoolean("Enable Zombie Drops", category, true, "");
		EventHandlerCommon.enableZomVildrops = configMain.getBoolean("Enable Zombie Villager Drops", category, true, "");
		if(EventHandlerCommon.goldcoinMaxDropAmt < EventHandlerCommon.goldcoinMinDropAmt)
			EventHandlerCommon.goldcoinMaxDropAmt = EventHandlerCommon.goldcoinMinDropAmt;
		if(EventHandlerCommon.ironcoinMaxDropAmt < EventHandlerCommon.ironcoinMinDropAmt)
			EventHandlerCommon.ironcoinMaxDropAmt = EventHandlerCommon.ironcoinMinDropAmt;
		
		category = "World Generation";
		configMain.addCustomCategoryComment(category, "World Generation Settings");
		genCoinsInWorld.goldChance = configMain.getInt("Chance of Goldcoins appearing in chests", category, 5, 0, 100, "");
		genCoinsInWorld.ironChance = configMain.getInt("Chance of Ironcoins appearing in chests", category, 5, 0, 100, "");
		genCoinsInWorld.minIronGen = configMain.getInt("Minimum amount of Ironcoins that can appear in chests.", category, 0, 0, 64, "");
		genCoinsInWorld.maxIronGen = configMain.getInt("Maximum amount of Ironcoins that can appear in chests.", category, 5, 0, 64, "");
		genCoinsInWorld.minGoldGen = configMain.getInt("Minimum amount of Goldcoins that can appear in chests.", category, 0, 0, 64, "");
		genCoinsInWorld.maxGoldGen = configMain.getInt("Maximum amount of Goldcoins that can appear in chests.", category, 5, 0, 64, "");
		
	}
}
